### DISCUSSION TITLE
Webpack outputting weird/uncorrect css

### DISCUSSION THREAD

**Discussion by tsClauz** [webpack/webpack#15875] [General]
# Bug report
Webpack outputs uncorrect and very weird css code which is unreadable by browser and doesn't get rendered

**What is the current behavior?**
with this configuration:
```
module.exports = {
    mode: 'production',
    entry: /* object of entries */,
    output: { 
        path: path.resolve(__dirname, `dashboard.snake-bot.xyz/__assets`),
        filename: '[name].js',
        publicPath: ''
    }, 
    module: {
        rules: [{ 
            test: /(\.jsx|\.js|.ts|.tsx)$/, 
            include: path.resolve(__dirname, `src`),
            use: { 
                loader: 'babel-loader', 
                options: {
                    presets: [["@babel/preset-react", {"runtime": "automatic"}], "@babel/preset-env", "@babel/preset-typescript"],
                    plugins: ["@babel/plugin-syntax-top-level-await"]
                }
            }
        },
        { 
            test: /\.css$/, 
            include: path.resolve(__dirname, `src`),
            use: ['style-loader', 'css-loader']
        }]
    }, 
    resolve: { 
        modules: [__dirname + '/node_modules'], 
        extensions: ['.jsx', '.js', '.ts', '.tsx'], 
    },
    experiments: {
        futureDefaults: true,
        topLevelAwait: true 
    },
}
```

and providing simple css such as
```
.bolded { font-weight: bold }
.italic { font-style: italic }
.strike { text-decoration: line-through }
```
and trying to import it in a .jsx/.tsx file to style elements, causes webpack to compile them with uncorrect code;
```

      import API from "!../../node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js";
      import domAPI from "!../../node_modules/style-loader/dist/runtime/styleDomAPI.js";
      import insertFn from "!../../node_modules/style-loader/dist/runtime/insertBySelector.js";
      import setAttributes from "!../../node_modules/style-loader/dist/runtime/setAttributesWithoutAttributes.js";
      import insertStyleElement from "!../../node_modules/style-loader/dist/runtime/insertStyleElement.js";
      import styleTagTransformFn from "!../../node_modules/style-loader/dist/runtime/styleTagTransform.js";
      import content, * as namedExport from "!!../../node_modules/css-loader/dist/cjs.js!./general.css";
      
      

var options = {};

options.styleTagTransform = styleTagTransformFn;
options.setAttributes = setAttributes;

      options.insert = insertFn.bind(null, "head";
    
options.domAPI = domAPI;
options.insertStyleElement = insertStyleElement;

var update = API(content, options;



export * from "!!../../node_modules/css-loader/dist/cjs.js!./general.css";
       export default content && content.locals ? content.locals : undefined;

head{--webpack-826:_714;}
```

**If the current behavior is a bug, please provide the steps to reproduce.**
- create the webpack.config.js
- create a .jsx file
- create a .css file
- try to import the .css file into the jsx
- compile
- observe the code 

**What is the expected behavior?**
The .css compiled code is readable by the browser and gets rendered

**Other relevant information:**
webpack version:  5.72.1
Node.js version: 18.2.0
Operating System: Debian 10
Additional tools:

**Comment by alexander-akait** [comment]
Webpack outputs JS to create runtime for CSS, you have this problem due invalid regexp, so not valid loader applied

**Reply by alexander-akait** [reply]
If you need raw CSS or you want to write own loader and handle it, you can look at https://github.com/webpack-contrib/css-loader#exporttype

**Reply by tsClauz** [reply]
aside from creating my own loader, how should I "run" the css runtime to get valid css?

**Reply by tsClauz** [reply]
edit: with the new config the valid css code is being outputted, but im getting `Uncaught (in promise) TypeError: s[e] is not a function` in browser if I try to import that css into my jsx (in short webpack compiles errored code if i import css into jsx)

**Reply by alexander-akait** [reply]
> aside from creating my own loader, how should I "run" the css runtime to get valid css?

What do you want to solve?

> edit: with the new config the valid css code is being outputted, but im getting Uncaught (in promise) TypeError: s[e] is not a function in browser if I try to import that css into my jsx (in short webpack compiles errored code if i import css into jsx)

Sounds like you try to run functions on CSS runtime, but don't have this runtime

**Reply by tsClauz** [reply]
that's what I don't get, what is the runtime and how to "run" that,
```
import '../css/general.css'

export function myElement() {
    return <p id="welcome"><span className="bolded">Hi</span></p>
}
```
this code when compiled throws that error, here I am importing general.css to get the class "bolded" styled, is there anything I should do differently?

**Reply by tsClauz** [reply]
oh lol im just stupid, didn't add the css into the .html file

**Reply by alexander-akait** [reply]
:smile: Solved?
